File: JavaSolutions/SumOfDigits.java
/**
Given a positive integer, find the sum of its constituent digits.

INPUT SAMPLE:

The first argument will be a path to a filename containing positive integers, one per line. E.g.

23
496
OUTPUT SAMPLE:

Print to stdout, the sum of the numbers that make up the integer, one per line. E.g.

5
19
*/
import java.io.*;

public class SumOfDigits {
  public static void main(String [] args) throws IOException{

    //try-catch to open the file provided by ARGS
    try {
      FileReader fr = new FileReader(args[0]);
      BufferedReader br = new BufferedReader(fr);
      String ln;
      while ((ln = br.readLine()) != null){
        int sum = 0;
        String [] ar = ln.split("");
        if (ar.length > 0){
          for (String e : ar){
            sum += Integer.parseInt(e);
          }
        }
        System.out.println(sum);
      }

      br.close();
    }
    catch(FileNotFoundException e){
      System.out.println("Could not open file.");
    }
  }
}

File: JavaScriptSolutions/sumofdigits.js
/*Given a positive integer, find the sum of its constituent digits.

INPUT SAMPLE:

The first argument will be a path to a filename containing positive integers, one per line. E.g.

23
496
OUTPUT SAMPLE:

Print to stdout, the sum of the numbers that make up the integer, one per line. E.g.

5
19

*/

let fs  = require("fs");

function summation(num){
  let numbers = [];
  let sum = 0;
  for (let n = 0; n < num.length; n++) {
    numbers[n] = parseInt(num.charAt(n));
  }
  for (let i = 0; i < num.length; i++){
    sum += numbers[i];
  }
  console.log(sum);
};

fs.readFileSync(process.argv[2]).toString().split('\n').forEach(function (line) {
    let sum = 0;
    if (line !== "") { // ignore empty lines
      summation(line);
    }
});
